vendor/github.com/MustWin/baremetal-sdk-go/request_options.go

```diff
@@ -259,7 +259,7 @@ type UpdateDHCPDNSOptions struct {
 type UpdateGatewayOptions struct {
 	IfMatchOptions
 	DisplayNameOptions
-	IsEnabled bool `header:"-" json:"isEnabled,omitempty" url:"-"`
+	IsEnabled *bool `header:"-" json:"isEnabled,omitempty" url:"-"`
```

Comment:
Should this change be in here?

Reply:
Yes, booleans values can only mean true or false, but boolean pointers can mean true, false or nil. True or false need to be transmitted as named json fields in the request, nil needs to be omitted from request so the current or default value is unaltered.